### PR TITLE
Improvements of Anime-Planet lists import

### PR DESCRIPTION
**Pull request by hohMiyazawa**
This looks promising. What's the status of it, @nzoba?

### CODE REVIEW

**Comment by nzoba** [comment]
It still needs a few improvements but it should be done by the end of the week-end or when I find time.
Your implementation saved me from manually importing hundreds of mangas, thanks a lot !
I'll tell you when it's ready

**Comment by nzoba** [comment]
It's done if you want to review it @hohMiyazawa 
I essentially added the possibility to chose among a max of 5 results for each title, to download the errors and search for only a particular format. Also I added the cover for each result and fixed some minor bugs.

File: src/utilities/levDist.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 function levDist(s,t){//https://stackoverflow.com/a/11958496/5697837
 	// Step 1
+	s = s.replace("’", "'")
+	t = t.replace("’", "'")
```
Comment (hohMiyazawa) [inline src/utilities/levDist.js:4]: Could have been done by replacing *before* passing the strings to leveDist(), but ignoring similar characters is probably best anyway.

**Comment by hohMiyazawa** [comment]
Looks good to me

**Comment by hohMiyazawa** [comment]
If you found some missed matches when importing, you can also add them to ``/src/data/AnimePlanet_mappings.json`` if they actually have an anilist entry.

File: src/utilities/levDist.js (hunk 1 above)
Comment (nzoba) [reply src/utilities/levDist.js:4]: Yes I thought so too but like you said I concluded that always ignoring these characters was the best way

**Comment by nzoba** [comment]
Oh right I forgot to do that. But I think that differentiating manga and anime would be better because both can have the same name and if everything is in one file the mapping would be wrong for one of them.

**Comment by hohMiyazawa** [comment]
Ah, good point.
Easiest way of doing that is probably to make manga a separate file, keeping the structure.

**Comment by hohMiyazawa** [comment]
Proposed solution, in code:

https://github.com/hohMiyazawa/Automail/commit/bc361507872ad178335b930e90e34d2f6de10c97
https://github.com/hohMiyazawa/Automail/commit/dadc61456995ed5c248b63167da73b1e0320f7ee

**Comment by nzoba** [comment]
I would have done something like that :  `const apMappings = type === "manga" ? m4_include(data/AnimePlanet_mappings_manga.json) : m4_include(data/AnimePlanet_mappings_anime.json);`

It only needs 1 line and is more transparent (or a bit more if you prefer if/else), especially since the logic is exactly the same for either file

**Comment by hohMiyazawa** [comment]
Mainly a style difference, and perfectly fine too.
I leave it completely up to pick that way if you decide to make some other improvements to the importer.

**Pull request merged**